### PR TITLE
Fix issue with behavior on rotation

### PR DESCRIPTION
**Pull request by Astach**
* Use Events Wrapper in order to have catch event only once with live data
* Do not initialize login controller when not needed.

Before this commit - 
* on device rotation user was redirected to the first screen because the ClientResult was emitted again (which cause a navigation to the first screen as intended).
* LoginController was created no matter what, and then the signup flow wasn't working (because handled by the login controller implementation, that means when you were trying to create an account under the hood the app was trying to log you in, and then you got an error saying "Sorry bro, credentials of yours are wrong".

### CODE REVIEW

File: ui/src/main/java/com/schibsted/account/ui/login/LoginActivityViewModel.kt [hunk 1]
```diff
@@ -105,24 +108,27 @@ class LoginActivityViewModel(
         if (intentClientInfo == null) {
             fetchClientInfo()
         } else {
-            clientResult.value = ClientResult.Success(intentClientInfo)
+            clientResult.setValue(Event(ClientResult.Success(intentClientInfo)))
```
Comment (okacat) [inline ui/src/main/java/com/schibsted/account/ui/login/LoginActivityViewModel.kt]: Why was this changed to use `setValue`?

File: ui/src/main/java/com/schibsted/account/ui/login/LoginActivityViewModel.kt [hunk 2]
```diff
@@ -131,11 +137,8 @@ class LoginActivityViewModel(
         }
     }
 
-    fun isFlowReady(): LiveData<Boolean?> {
-        if (smartlockCredentials.value == null && smartlockReceiver.isSmartlockResolving.value) {
-            return MutableLiveData<Boolean>().apply { value = userFlowType == null }
-        }
-        return MutableLiveData<Boolean>().apply { value = null }
+    fun isFlowReady(): Boolean {
```
Comment (okacat) [inline ui/src/main/java/com/schibsted/account/ui/login/LoginActivityViewModel.kt]: Could be shortened by using `=`

File: ui/src/main/java/com/schibsted/account/ui/login/LoginActivityViewModel.kt (hunk 1 above)
Comment (Astach) [reply ui/src/main/java/com/schibsted/account/ui/login/LoginActivityViewModel.kt]: Ahah it's just that I forgot to change it. During my refactoring, I reach a state where `ClientResult` was not a `MutableLiveData` anymore but another object with a `setValue` method :sweat:
